### PR TITLE
Create reverse_linked_list.py

### PR DESCRIPTION
**Pull request by ManishKumar219**
### Describe your change:

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: data_structures/linked_list/reverse_linked_list.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+class Node:
+
```
Comment (CaedenPH) [inline data_structures/linked_list/reverse_linked_list.py:2]:
```suggestion
```

File: data_structures/linked_list/reverse_linked_list.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    # Constructor to initialize the node object
+    def __init__(self, data):
+        self.data = data
+        self.next = None
+
+
+class LinkedList:
+
```
Comment (CaedenPH) [inline data_structures/linked_list/reverse_linked_list.py:10]:
```suggestion
```

File: data_structures/linked_list/reverse_linked_list.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+llist.push(20)
+llist.push(4)
+llist.push(15)
+llist.push(85)
+
+print("Given Linked List")
+llist.printList()
+llist.reverse()
+print("\nReversed Linked List")
+llist.printList()
```
Comment (CaedenPH) [inline data_structures/linked_list/reverse_linked_list.py:51]:
```suggestion
if __name__ == "__main__":
    # Driver program to test above functions
    llist = LinkedList()
    llist.push(20)
    llist.push(4)
    llist.push(15)
    llist.push(85)
    
    print("Given Linked List")
    llist.printList()
    llist.reverse()
    print("\nReversed Linked List")
    llist.printList()
 ```

File: data_structures/linked_list/reverse_linked_list.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        self.head = prev
+
+    # Function to insert a new node at the beginning
+    def push(self, new_data):
+        new_node = Node(new_data)
+        new_node.next = self.head
+        self.head = new_node
+
+    # Utility function to print the LinkedList
+    def printList(self):
```
Comment (CaedenPH) [inline data_structures/linked_list/reverse_linked_list.py:33]:
```suggestion
    def print_list(self):
```
Python uses `snake_case`

File: data_structures/linked_list/reverse_linked_list.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        current = self.head
+        while current is not None:
+            next = current.next
+            current.next = prev
+            prev = current
+            current = next
+        self.head = prev
+
+    # Function to insert a new node at the beginning
+    def push(self, new_data):
```
Comment (CaedenPH) [inline data_structures/linked_list/reverse_linked_list.py:27]:
```suggestion
    def push(self, new_data) -> None:
```

File: data_structures/linked_list/reverse_linked_list.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+
+class LinkedList:
+
+    # Function to initialize head
+    def __init__(self):
+        self.head = None
+
+    # Function to reverse the linked list
+    def reverse(self):
```
Comment (CaedenPH) [inline data_structures/linked_list/reverse_linked_list.py:16]:
```suggestion
    def reverse(self) -> None:
```

File: data_structures/linked_list/reverse_linked_list.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    # Constructor to initialize the node object
+    def __init__(self, data):
+        self.data = data
+        self.next = None
+
+
+class LinkedList:
+
+    # Function to initialize head
+    def __init__(self):
```
Comment (CaedenPH) [inline data_structures/linked_list/reverse_linked_list.py:12]:
```suggestion
    def __init__(self) -> None:
```

File: data_structures/linked_list/reverse_linked_list.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
+class Node:
+
+    # Constructor to initialize the node object
+    def __init__(self, data):
```
Comment (CaedenPH) [inline data_structures/linked_list/reverse_linked_list.py:4]:
```suggestion
    def __init__(self, data) -> None:
```

**Review by CaedenPH — CHANGES_REQUESTED** [review]
Please could you add [doctests](https://docs.python.org/3/library/doctest.html)

In addition, instead of printing the reverse linked list, would it be possible for you to return a reversed tree?

**Comment by cclauss** [comment]
Duplicate https://github.com/TheAlgorithms/Python/blob/master/data_structures/linked_list/print_reverse.py

**Pull request closed without merging**